### PR TITLE
CVE-2015-1292 and CVE-2014-7908

### PR DESCRIPTION
**Pull request by dis446**
Investigations into two Chromium vulnerabilities.

### CODE REVIEW

File: cves/CVE-2014-7908.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
 announced_instructions: |
```
Comment (andymeneely) [inline cves/CVE-2014-7908.yml]: Looks like you came across one that was already curated in the past. Don't just leave this one alone - make it better!

Comment (dis446) [reply cves/CVE-2014-7908.yml]: Okay. Will do. Sorry I've had a very busy weekend. Will start on that tomorrow.

File: cves/CVE-2015-1292.yml [hunk 2]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
...
   If you are manually editing this file, then you are "curating" it. Set the
   entry below to "true" as soon as you start. This will enable additional
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
 announced_instructions: |
   Was there a date that this vulnerability was announced to the world? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE date. A good
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: Remove timestamp from "announced" below (keep date)

File: cves/CVE-2015-1292.yml [hunk 3]
```diff
@@ -100,20 +105,25 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: Based on the description in the CVE and blog post.
+  name: webKit
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: Capitalize WebKit

File: cves/CVE-2015-1292.yml [hunk 4]
```diff
@@ -159,38 +172,42 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Just because someone knows how to request access to a resource does not mean
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: Reword to answer in the context of this specific vulnerability

File: cves/CVE-2015-1292.yml [hunk 5]
```diff
@@ -159,38 +172,42 @@ lessons:
...
+      that they are necessarily authorized to access it.
   least_privilege:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Only requests that come from the same origin should be granted access to resources. No others.
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: Same note as above

File: cves/CVE-2015-1292.yml [hunk 6]
```diff
@@ -159,38 +172,42 @@ lessons:
...
-    applies: 
-    note: 
+    applies:
+    note:
   security_by_obscurity:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      If no one knew about this weird behaviour, it would not have ever been a problem
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: What is the weird behavior you're referring to? Add more detail.

File: cves/CVE-2015-1292.yml [hunk 7]
```diff
@@ -203,4 +220,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The coding mistake made here was being too excited in implementing a new feature
+    and not checking that the entities using the feature are who they say they are.
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-1292.yml]: Would this be a design mistake, a coding mistake, or both?

File: cves/CVE-2015-1292.yml (hunk 6 above)
Comment (dis446) [reply cves/CVE-2015-1292.yml]: Okay. Will do. By weird behavior i meant if no one knew about the defect, no exploits would occur.

File: cves/CVE-2015-1292.yml (hunk 2 above)
Comment (dis446) [reply cves/CVE-2015-1292.yml]: Will do. Guess the time doesn't really matter

File: cves/CVE-2015-1292.yml (hunk 7 above)
Comment (dis446) [reply cves/CVE-2015-1292.yml]: Design mostly. I add notes about it

File: cves/CVE-2014-7908.yml [hunk 8]
```diff
@@ -34,7 +34,9 @@ description_instructions: |
   keep too.
 description: |
   Integer overflow for multiple media file types. This overflow vulnerability
-  allows attackers to cause a denial of serivce.
+  allows attackers to cause a denial of serivce. Any Mov/QuickTime/MPEG4 Container
+  had a defect that could cause performance degradation and potentially be used to
```
Comment (KevinWongRIT) [inline cves/CVE-2014-7908.yml]: Either "Any Mov/QuickTime/MPEG4 Container _that_ had a defect -that- could cause..." or "_All_ Mov/QuickTime/MPEG4 Container_s_ had a defect -that- could cause..."

File: cves/CVE-2014-7908.yml [hunk 9]
```diff
@@ -119,8 +121,10 @@ interesting_commits:
     note: Third attempt of fixing narrowing error for TAG. There were several commits
       to this file attempting to fix another bug. This is interesting because this
       additional bug was not easily fixed.
-  - commit: 
-    note: 
+  - commit: c9533cf7f87f8511f20e933e4c42ed52184dd8d9
+    note: Attempted to fix a crash in the container names CheckMov function (the same function where the
```
Comment (KevinWongRIT) [inline cves/CVE-2014-7908.yml]: Did this fix work?  If not, why?

File: cves/CVE-2015-1292.yml (hunk 2 above)
Comment (KevinWongRIT) [reply cves/CVE-2015-1292.yml]: One of the example YAMLs given included the time while the other didn't so I'm not sure it really matters.

File: cves/CVE-2015-1292.yml [hunk 10]
```diff
@@ -159,38 +172,43 @@ lessons:
...
-    applies: 
-    note: 
+    applies:
+    note:
   security_by_obscurity:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      If no one knew about this weird behaviour, it would not have ever been a problem
```
Comment (KevinWongRIT) [inline cves/CVE-2015-1292.yml]: This is really nitpicky but it is spelled "behavior" in American English.

File: cves/CVE-2015-1292.yml (hunk 6 above)
Comment (KevinWongRIT) [reply cves/CVE-2015-1292.yml]: The lesson of security by obscurity is that it should not be relied on.

File: cves/CVE-2015-1292.yml [hunk 11]
```diff
@@ -100,20 +105,25 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: Based on the description in the CVE and blog post.
+  name: WebKit
```
Comment (KevinWongRIT) [inline cves/CVE-2015-1292.yml]: Isn't the file path of the source code under blink rather than webkit?

File: cves/CVE-2015-1292.yml [hunk 12]
```diff
@@ -60,16 +65,16 @@ fixes:
...
+  :note: 'This defect has been in webKit since they added service Workers'
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes:
```
Comment (KevinWongRIT) [inline cves/CVE-2015-1292.yml]: +1

File: cves/CVE-2014-7908.yml [hunk 13]
```diff
@@ -11,7 +11,7 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
```
Comment (KevinWongRIT) [inline cves/CVE-2014-7908.yml]: +3 for upvotes

File: cves/CVE-2015-1292.yml (hunk 12 above)
Comment (Inchworm333) [reply cves/CVE-2015-1292.yml]: +2

File: cves/CVE-2015-1292.yml [hunk 14]
```diff
@@ -159,38 +172,43 @@ lessons:
...
+    note: |
+      Just because someone knows how to request access to a service does not mean
+      that they are necessarily authorized to get one. If they get a service worker 
+      then they can use it to inject their own code.
   least_privilege:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Only requests that come from the same origin should be granted access to resources, specifically . No others.
```
Comment (Inchworm333) [inline cves/CVE-2015-1292.yml]: What about for the usage of Chrome extensions?

**Pull request merged**